### PR TITLE
SvelteKit release candidate migration: links

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
- Moved [`/src/lib/constants/links.ts`](https://github.com/EthanThatOneKid/acmcsuf.com/blob/36d15b5cdf373b3c594c773ebf3b3942e54677c6/src/lib/constants/links.ts) to `/src/lib/data/`
- Copied features from #450
- Migrated link shortener hook to `/src/hooks.server.ts`

See: <https://kit.svelte.dev/docs/hooks#server-hooks>

### CODE REVIEW

File: src/routes/[link=link].svg/+server.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import type { RequestHandler } from './$types';
+import { LINKS } from '$lib/links/data';
+import { parseLink, genQRCodeSvg } from '$lib/links/utils';
+
+export const GET = handler();
+
+function handler(): RequestHandler {
+  return async function (event) {
+    const link = parseLink(event.params.link);
```
Comment (EthanThatOneKid) [inline src/routes/[link=link].svg/+server.ts]: Nit: Naming is confusing.
```suggestion
    const id = parseLinkId(event.params.link);
```

Comment (EthanThatOneKid) [reply src/routes/[link=link].svg/+server.ts]: Done.

**Review by anguzz — COMMENTED** [review]
Amazing, qr code feature still looking good☑

**Review by anguzz — COMMENTED** [review]
looks good!

**Review by anguzz — APPROVED** [review]

**Comment by EthanThatOneKid** [comment]
Replaced by mega-PR #577.

**Pull request closed without merging**